### PR TITLE
[ssu] Allow retrieving and setting the domain via DBus. Contributes t…

### PR DESCRIPTION
**Pull request by giucam**
…o JB#16892

### CODE REVIEW

File: ssud/ssud.h [hunk 1]
```diff
@@ -32,6 +32,8 @@ class Ssud: public QObject {
     bool isRegistered();
     void registerDevice(const QString &username, const QString &password);
     void unregisterDevice();
+    void setDomain(const QString &domain);
+    QString currentDomain();
```
Comment (saukko) [inline ssud/ssud.h]: ooi, shouldn't this be const? :)

**Comment by bwachter** [comment]
The only supported way to change the domain is by doing a new registration. For the cases where people change the domain for testing purposes the current CLI should be fine. There should be no way for people without a proper understanding of how device registration works to accidentally change the domain.

**Comment by giucam** [comment]
Ok, so i'll remove the setDomain() and keep the currentDomain() only.

File: ssud/ssud.h (hunk 1 above)
Comment (giucam) [inline ssud/ssud.h]: It can't be, because it has to restart the autoclose timer.

File: dbus/org.nemo.ssu.xml [hunk 2]
```diff
@@ -35,6 +35,9 @@
     <method name="isRegistered">
       <arg direction="out" type="b" name="status"/>
     </method>
+    <method name="currentDomain">
+        <arg direction="out" type="s" name="domain"/>
+    </method>
```
Comment (saukko) [inline dbus/org.nemo.ssu.xml]: I wonder if this should be just domain instead of currentDomain, we have also e.g. flavour there and not currentFlavour.

**Comment by saukko** [comment]
:+1:

**Pull request merged**